### PR TITLE
Raise error for missing sample data path.

### PR DESCRIPTION
**Pull request by rhattersley**
See https://groups.google.com/forum/#!topic/scitools-iris/gv7vB7VGH9M

Ping @ajdawson

### CODE REVIEW

File: lib/iris/__init__.py [hunk 1]
```diff
@@ -427,4 +428,10 @@ def load_raw(uris, constraints=None, callback=None):
 
 def sample_data_path(*path_to_join):
     """Given the sample data resource, returns the full path to the file."""
-    return os.path.join(iris.config.SAMPLE_DATA_DIR, *path_to_join)
+    target = os.path.join(iris.config.SAMPLE_DATA_DIR, *path_to_join)
+    if not glob.glob(target):
```
Comment (pelson) [inline lib/iris/__init__.py]: Can we go even further. Even if it does result in files, can we prevent the following from happening:

```
sample_data_path('/absolute', 'path')
```

I'm afraid to say I have seen this in the wild, and it upset me. :cry:

Comment (pelson) [inline lib/iris/__init__.py]: Just in case you were wondering:

```
>>> os.path.join('foo', '/bar', 'wibble')
'/bar/wibble'

```

Comment (rhattersley) [inline lib/iris/__init__.py]: Yeah, I mucked around with this for a short while but the definition of "wrong" was somewhat ambiguous. For example, if I've installed the sample data at `/samples` then is it unambiguously wrong to do `sample_data_path('/samples/air_temp.pp')`? It's certainly against the _spirit_ of the function. It probably wouldn't take too much to persuade me to rule out _any_ absolute path - at which point we've probably caught most of the common misuse patterns.

Comment (ajdawson) [inline lib/iris/__init__.py]: The whole point of the function is to be able to supply a relative path and not have to care about the absolute path to the samples directory, so absolute paths have no place in the arguments to this function. If someone is doing `sample_data_path('/full/path/to/samples/air_temp.pp')` then they are misusing this function. The new exception should discourage them from such behaviour!

**Comment by ajdawson** [comment]
Let's also stick in a check for absolute paths in the input and have done with it. Doing sure will tighten down the behavioral scope of the function to its intended use. Otherwise I think this is good, the exception is clear and should act to prevent misuse of this function.

**Comment by rhattersley** [comment]
> Let's also stick in a check for absolute paths in the input and have done with it.

Like I said ... it didn't take much to persuade me. :wink:

**Comment by pelson** [comment]
Good call @ajdawson, and thanks for doing it @rhattersley. 👍

**Pull request merged**